src/main/java/graphql/schema/GraphQLArgument.java

```diff
@@ -56,10 +62,26 @@ public GraphQLInputType getType() {
         return type;
     }
 
+    /**
+     * An argument has a default value when it represents the logical argument structure that a {@link graphql.schema.GraphQLFieldDefinition}
+     * can have and it can also have a default value when used in a schema definition language (SDL) where the
+     * default value comes via the directive definition.
+     *
+     * @return the default value of an argument
+     */
     public Object getDefaultValue() {
         return defaultValue;
     }
 
+    /**
+     * An argument ONLY has a value when its used in a schema definition language (SDL) context as the arguments to SDL directives.  The method
+     * should not be called in a query context, but rather the AST / variables map should be used to obtain an arguments value.
+     *
+     * @return the argument value
+     */
+    public Object getValue() {
+        return value;
+    }
 
```

Comment:
The above is the possible contentious change. 

It comes about because a argument can have a "possible default value" and an actual value AND in 2 different contexts.

When we have a query like

    query {
        field (arg : "123", arg2 :$var)

then the argument values ONLY ever come from the AST literal or variables map.

But in DSL directives, we have an actual static value and NO variables map.

    type Query {
        field : String @directive(withArg : "argValue")
    }

The above can also have a default from a possible directive definition elsewhere is SDL.

I chose to have it this way after debating naming and so on.  Previously a made getDefaultValue() hold the actual value but this is semantically wrong.  I can see now that it was wrong.

So I created the getValue() method but know its never called in "query context".   It could have been called "getDirectValue()" or some such but I couldnt make up a name that made it any clearer so I opted for the most straight forward name